evennia/server/profiling/profiler.py

```diff
@@ -0,0 +1,177 @@
+import gc
+import inspect
+import tracemalloc
+import linecache
+import sys
+from datetime import datetime
+from functools import wraps
+
+TRASH = [
+    tracemalloc.Filter(False, "<frozen importlib._bootstrap>"),
+    tracemalloc.Filter(False, "<frozen importlib._bootstrap_external>"),
+    tracemalloc.Filter(False, "<frozen ntpath>"),
+    tracemalloc.Filter(False, "<frozen abc>")
+]
+
+KB = 1024
+MB = 1024*1024
+GB = 1024*1024*1024
+
+# snaps = []
+
+# def snapshot():
+#     snaps.append(tracemalloc.take_snapshot())
+
+# def trace(func, stream=None, frames=None):
+#     if stream is None:
+#         stream = sys.stdout
+
+#     @wraps(func)
+#     def wrapper(*args, **kwargs):
+#         if not tracemalloc.is_tracing():
+#             if frames:
+#                 tracemalloc.start(frames)
+#             else:
+#                 tracemalloc.start()
+
+#         result = func(*args, **kwargs)
+
+#         collect()
+#         snapshot()
+#         tracemalloc.stop()
+#         top(stream)
+
+#         return result
+
+#     return wrapper
```

Comment:
Leftover code?

Reply:
I didn't get the function wrapper to ever get the right frame. I think I could fix this now, but at least for the next week or so my priorities will be elsewhere. I can just remove this for now and add it later.